tests/ui/type-alias-impl-trait/self-referential-3.rs

```diff
@@ -1,9 +1,9 @@
-// run-pass
 #![feature(type_alias_impl_trait)]
 
 type Bar<'a, 'b> = impl PartialEq<Bar<'a, 'b>> + std::fmt::Debug;
 
 fn bar<'a, 'b>(i: &'a i32) -> Bar<'a, 'b> {
+    //~^ ERROR can't compare `&i32` with `Bar<'a, 'b>`
```

Comment:
why does this break? :thinking: is `Bar<'a, 'b>` a `ty::Opaque` here or a weak projection?

Reply:
I opened https://github.com/rust-lang/rust/pull/109115

Reply:
We could also just run `normalize_with_depth` on the nested obligations in the assoc type normalizer, like the rest of the selection/etc. code does